src/arrayMethodJoin.js

```diff
@@ -5,7 +5,31 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    if (this.length === 0) {
+      return '';
+    }
+
+    const sourceAarrayCopy = [...this];
```

Comment:
Dont need to copy array [checklist](https://github.com/mate-academy/js_array-method-join/blob/master/checklist.md) pt. 2 (main regect reason)